code/clue.js

```diff
@@ -7,53 +7,267 @@ const mrGreen = {
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+const delayedPickKiller = () => {  // This will randomly select a killer from the suspects. And add that to the mystery object.
+  document.getElementById('loader1').style.display = 'none';
+  document.getElementById("killerCard").removeEventListener("click", pickKiller); //so that the card can't be clicked again and trigger the loader
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+  if(mystery.killer === undefined) {   //To check if card has been clicked already or not - undefined= not clicked
+    mystery.killer = randomSelector(suspects);
+    mystery.killer.favouriteWeapon = randomSelector(weapons);  //Randomly reassigns killer's favorite weapon
+      
+    // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+    document.getElementById('killerCard').style.background = mystery.killer.color;
+    document.getElementById('killerName').innerHTML = `Full name: ${mystery.killer.firstName} ${mystery.killer.lastName}`;
+    document.getElementById('killerAge').innerHTML = `Age: ${mystery.killer.age}`;
+    document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`;
+    document.getElementById('killerImage').src = `${mystery.killer.image}`;
+    document.getElementById('killerDescription').innerHTML = `Description: ${mystery.killer.description}`;
+    document.getElementById('killerFavWeapon').innerHTML = `Favourite Weapon: ${mystery.killer.favouriteWeapon.name}`; //Display killer's fav weapon on HTML
+  }
+ };
+
+// PICK WEAPON FUNCTION - This function will be invoked when you click on the weapon card.
+document.getElementById("weaponCard").addEventListener("click", pickWeapon);
+
+function pickWeapon() { 
+  document.getElementById('loader2').style.display = 'block';
+  setTimeout(delayedPickWeapon, 1500);
+};
+
+const delayedPickWeapon = () => {
+  document.getElementById('loader2').style.display = 'none';
+  document.getElementById("weaponCard").removeEventListener("click", pickWeapon);
+
+  // Red and Black level functions for generating a favotire weapon
+  if(mystery.weapon === undefined) { //To check if card has been clicked already or not - undefined= not clicked
+    const modifiedWeaponsArray = weapons; // Creates new Array to add killer's favorite weapon several more times so chances are higher it is picked
+
+    if(mystery.killer !== undefined) { //In case weapon card is clicked before a killer is defined, so code doesn't break. If killer has not been defined, it will generate a weapon without taking into account the killer's fav weapon. If a killer is defined, we know their fav weapon and cann add it to the array
```

Comment:
Putting the comments above the line is a commonly accepted convention which sometimes aides readability 
```javascript
/*In case weapon card is clicked before a killer is defined, 
 * so code doesn't break. If killer has not been defined, 
 * it will generate a weapon without taking into account the killer's fav weapon. 
 * If a killer is defined, we know their fav weapon and cann add it to the array
 */
    if(mystery.killer !== undefined) { 
```